### PR TITLE
Bug 1728809 - Implement a simple Node.js sample app

### PR DESCRIPTION
**Pull request by brizental**
~This is a draft until https://github.com/mozilla/glean_parser/pull/390 makes it into a release and `glean_parser` is updated here.~ https://github.com/mozilla/glean_parser/releases/tag/v4.1.0

### Pull Request checklist ###

- [x] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] ~**Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one~
- [x] **Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work

### CODE REVIEW

File: samples/node/main.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+"use strict";
+
+import Glean from "@mozilla/glean/node";
+import { custom } from "./generated/pings.js";
+import { appStarted } from "./generated/sample.js";
+
+export default () => {
+  Glean.setLogPings(true);
+  // Uncomment when you need to inspect the pings sent in the Glean Debug Ping Viewer.
+  // Glean.setDebugViewTag("gleanjs-node-sample");
```
Comment (badboy) [inline samples/node/main.js:14]: Make it depend on an app argument or an environment variable!

**Review by badboy — APPROVED** [review]
This is so boring compared to the 300 lines of code it takes for our iOS/Android sample app!

File: samples/node/main.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const testAppId = `node.test.${this.title}`;
+
+  beforeEach(async function () {
+    await Glean.testResetGlean(testAppId);
+  });
+
+  describe("sample test", function () {
+    it("appStarted metric is recorded and ping is sent", async function () {
+      // Setup validator for sending of the custom ping.
+      const pingWasSent = custom.testBeforeNextSubmit(async () => {
```
Comment (badboy) [inline samples/node/main.test.js:22]: Yey, big plus for having tests on the sample app!

File: samples/node/main.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        // Check no errors were recorded on that metric either.
+        assert.equal(
+          await appStarted.testGetNumRecordedErrors(ErrorType.InvalidValue),
+          0
+        );
+      });
+
+      // Call the sample script.
+      // This will record the metric and send the ping we are expecting.
+      void main();
```
Comment (badboy) [inline samples/node/main.test.js:34]: huh, this looks like a function declaration in C.
Is the `void ...` pattern just for ignoring any return value?

Comment (brizental) [reply samples/node/main.test.js:34]: Exactly! But it is actually not necessary and a leftover from development here. For a time the `main` function used to return a Promise, so I put the `void` there so the linter would be fine with me not `await`ing on it. Let me remove it... 
```suggestion
      main();
```

File: samples/node/main.js (hunk 1 above)
Comment (brizental) [reply samples/node/main.js:14]: Huh, I guess since we are in Node.js land we _can_ use env vars right. That is a nice suggestion, I'll change it.

**Pull request merged**